lms/djangoapps/courseware/views/views.py

```diff
@@ -1026,6 +1028,29 @@ def program_marketing(request, program_uuid):
     return render_to_response('courseware/program_marketing.html', context)
 
 
+@ensure_csrf_cookie
+@ensure_valid_course_key
+def dates(request, course_id):
+    """
+    Display the course's dates.html, or 404 if there is no such course.
+    Assumes the course_id is in a valid format.
+    """
+
+    course_key = CourseKey.from_string(course_id)
+    course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+    course_date_blocks = get_course_date_blocks(course, request.user, request, include_past_dates=True)
+    learner_is_verified = get_enrollment(request.user.username, course_id)
```

Comment:
This feels like a weird variable name. I'm expecting a bool based on the name, but I assume you're getting an Enrollment object back. Is that correct or am I just off base here about the return type?

Reply:
Yeah I'll update this to check the actual mode, was treating existence as being verified which isn't accurate